virtualizarr/zarr.py

```diff
@@ -96,3 +106,67 @@ def ceildiv(a: int, b: int) -> int:
     See https://stackoverflow.com/questions/14822184/is-there-a-ceiling-equivalent-of-operator-in-python
     """
     return -(a // -b)
+
+
+def dataset_to_zarr(ds: xr.Dataset, storepath: str) -> None:
+    """
+    Write an xarray dataset whose variables wrap ManifestArrays to a Zarr store, writing chunk references into manifest.json files.
+
+    Not very useful until some implementation of a Zarr reader can actually read these manifest.json files.
+    See https://github.com/zarr-developers/zarr-specs/issues/287
+    """
+
+    from virtualizarr.manifests import ManifestArray
+
+    _storepath = Path(storepath)
+    Path.mkdir(_storepath, exist_ok=False)
+
+    # TODO should techically loop over groups in a tree but a dataset corresponds to only one group
+    # TODO does this mean we need a group kwarg?
+
+    consolidated_metadata: dict = {"metadata": {}}
+
+    # write top-level .zattrs
+    with open(_storepath / ".zattrs", "w") as json_file:
+        json.dump(ds.attrs, json_file, indent=4, separators=(", ", ": "))
```

Comment:
You might consider using `zarr.util.json_dumps` here. It would be a bit cleaner and more or less guarantees that you will be able to encode all valid zarr metadata. For example, it uses a custom number encoder to encode some dtypes.

Reply:
@jhamman cool - sounds like this package should have an optional dependence on zarr-python for utility functions like that then.

@cisaacstern why not? 😄 I won't put that in this PR, but it can be a follow-up (tracked by #46).